dc/s2n-quic-dc/src/path/secret/map/rehandshake.rs

```diff
@@ -30,6 +27,10 @@ impl RehandshakeState {
             handshake_at: Default::default(),
             schedule_handshake_at: Instant::now(),
             rehandshake_period,
+
+            // Initializes the hasher with random keys, ensuring we handshake with peers in a
+            // different, random order from different hosts.
+            hasher: RandomState::new(),
```

Comment:
do we need to rotate this state over time? the current behavior is the hosts that get the lowest hash values will always be prioritized over higher ones.

Reply:
I'm not sure what you mean by prioritized. We run through the full list before resuming, so while low (or high -- I never remember which order sorting goes in?) go first/last, I'm not sure that means much? The hash of a given peer address will be different on ~every endpoint (should be randomly mapped given a reasonably good hash, which SipHash seems to be).

I think rotating the state gets us to the random shuffling we were doing before, which is bad for p100 ages, and doesn't actually add a lot of value -- I can't think of why we'd expect rotating it to improve on something vs. not rotating, since they are randomly picked to start. Do you have a model for why shuffling each time would be better?